pennylane/_device.py

```diff
@@ -373,11 +394,22 @@ def check_validity(self, queue, observables):
         """
         for o in queue:
             if o.name not in self.operations:
-                raise DeviceError("Gate {} not supported on device {}".format(o.name, self.short_name))
+                raise DeviceError(
+                    "Gate {} not supported on device {}".format(o.name, self.short_name)
+                )
 
         for o in observables:
-            if o.name not in self.observables:
-                raise DeviceError("Observable {} not supported on device {}".format(o.name, self.short_name))
+            if isinstance(o, Observable) and (o.name not in self.observables):
+                raise DeviceError(
+                    "Observable {} not supported on device {}".format(o.name, self.short_name)
+                )
+
+            if isinstance(o, Tensor) and (self.capabilities()["tensor_observables"] is False):
+                raise DeviceError(
+                    "Tensor observables {} not supported on device {}".format(
+                        o.ops, self.short_name
+                    )
+                )
```

Comment:
Nice! We should start to make proper use of the capabilities dictionary. Other things we should declare are

* `formulation` (qubit or CV)
* `hardware` (true or false)
* `state_accessible` (true or false, for simulators that allow the state vector to be extracted)